rocketpy/simulation/dispersion.py

```diff
@@ -153,9 +156,260 @@ def __init__(
             self._error_file = f"{filename}.disp_errors.txt"
 
         # TODO: Initialize variables so they can be accessed by MATLAB
-        return None
 
-    # getters and setters for dispersion input/output/error files
+    # TODO move export_list to init
+    def run_dispersion(self, number_of_simulations, append=False):
+        """
+        Runs the dispersion simulation and saves all data.
+
+        Parameters
+        ----------
+        number_of_simulations : int
+            Number of simulations to be run, must be non-negative.
+        append : bool, optional
+            If True, the results will be appended to the existing files. If
+            False, the files will be overwritten. Default is False.
+
+        Returns
+        -------
+        None
+        """
+        # Create data files for inputs, outputs and error logging
+        open_mode = "a" if append else "w"
+        input_file = open(self._input_file, open_mode, encoding="utf-8")
+        output_file = open(self._output_file, open_mode, encoding="utf-8")
+        error_file = open(self._error_file, open_mode, encoding="utf-8")
+
+        # initialize counters
+        self.number_of_simulations = number_of_simulations
+        self.iteration_count = self.num_of_loaded_sims if append else 0
+        self.start_time = time()
+        self.start_cpu_time = process_time()
+
+        # Begin display
+        print("Starting monte carlo analysis", end="\r")
+
+        try:
+            while self.iteration_count < self.number_of_simulations:
+                self.__run_single_simulation(input_file, output_file)
+        except (TypeError, ValueError, KeyError, AttributeError) as error:
+            print(f"Error on iteration {self.iteration_count}: {error}")
+            error_file.write(f"{self._inputs_dict}\n")
+            self.__close_files(input_file, output_file, error_file)
+            raise error
```

Comment:
General exceptions are not so welcome as they tend to generalize they tend to generalize the behavior.
If an exception is raised without being in the list of expected exceptions, the code will still stop running, buuuut.... 
I'm curious to know what happens if you finish execution without calling the `self.__close_files(input_file, output_file, error_file)` line.

Reply:
I agree that a general Exception is not in general a good practice. I still need to read this code a bit better, but it would be basically to not stop the execution by a possibly "simple" error.

I am sorry if I am suggesting here something already implemented.

There are many solutions that we could employ to make the behavior of the dispersion in case of an exception convenient:

- Make a count incremented by each exception so that if too many simulations fail sequentially the program stops;

- Allow for saving partial dispersion results to be reused in a possible simulation restart.

Reply:
> because we need to detect if a simulation failed (basically by any reason).

I can agree with using the general exception then, given your comment.